File: solutions/_387_first_unique_character_in_a_string/solution.js
/**
 * @param {string} s
 * @return {number}
 */
var firstUniqChar = function (s) {
  const chars = new Map();
  for (const char of s) {
    chars.set(char, (chars.get(char) || 0) + 1);
  }

  for (let i = 0; i < s.length; i++) {
    if (chars.get(s[i]) === 1) {
      return i;
    }
  }
  return -1;
};

module.exports = { firstUniqChar };

/*
https://leetcode.com/problems/first-unique-character-in-a-string/
*/

File: solutions/_387_first_unique_character_in_a_string/solution.java
package _387_first_unique_character_in_a_string;

import java.util.*;

class Solution {
    public int firstUniqChar(String s) {
        Map<Character, Integer> map = new HashMap<>();
        for (char c : s.toCharArray()) {
            map.put(c, map.getOrDefault(c, 0) + 1);
        }

        for (int i = 0; i < s.length(); i++) {
            if (map.get(s.charAt(i)) == 1) {
                return i;
            }
        }

        return -1;
    }
}

/**
 * https://leetcode.com/problems/first-unique-character-in-a-string/
 */